File: InMotionGIT.Common/Events/ProgressEventArgs.vb
Namespace Events

    Public Class ProgressEventArgs
        Inherits EventArgs

        Property Message As String
        Property Level As Integer
        Property Index As Integer

        Public Sub New(ByVal message As String, ByVal level As Integer, ByVal index As Integer)
            Me.Message = message
            Me.Level = level
            Me.Index = index
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common/Events/ProgressEventArgs.cs
using System;

namespace InMotionGIT.Common.Events
{

    public class ProgressEventArgs : EventArgs
    {

        public string Message { get; set; }
        public int Level { get; set; }
        public int Index { get; set; }

        public ProgressEventArgs(string message, int level, int index)
        {
            Message = message;
            Level = level;
            Index = index;
        }

    }

}